CRM/Contribute/BAO/Query.php

```diff
@@ -568,6 +568,9 @@ public static function whereClauseSingle(&$values, &$query) {
 
       case 'contribution_batch_id':
         $batches = CRM_Contribute_PseudoConstant::batch();
+        // The key 'NULL' indicates that the user is looking for contributions
+        // that are not contained in a batch, see CRM-19325.
+        $batches['NULL'] = ts('(none)');
```

Comment:
@johanv Johan why not set here $op = 'IS NULL' and $value = '' if $batches = 'NULL' ? It would be similar to L345 of this file and then i think you wouldn't need the changes in the CRM_Contact_BAO_Query i don't think